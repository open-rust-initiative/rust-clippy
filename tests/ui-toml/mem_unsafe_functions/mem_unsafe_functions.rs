#![warn(clippy::mem_unsafe_functions)]
#![allow(unused)]

mod libc {
    pub fn safe() {}
    pub fn not_safe() {}
    pub fn memcpy() {}
}

extern "C" {
    fn safe();
    fn safe_1();
    fn not_safe();
    fn dont_use();
    fn del_mem();
    fn memcpy();
}

fn main() {
    unsafe {
        // Don't trigger
        safe();
        safe_1();
        libc::safe();
        libc::not_safe();
        libc::memcpy(); // because it's user defined

        // should trigger
        not_safe();
        dont_use();
        del_mem();
        memcpy();
    }
}
