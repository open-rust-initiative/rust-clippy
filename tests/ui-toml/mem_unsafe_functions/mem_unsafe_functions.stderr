error: use of potentially dangerous memory manipulation function
  --> $DIR/mem_unsafe_functions.rs:29:9
   |
LL |         not_safe();
   |         ^^^^^^^^
   |
   = help: consider using its safe version
   = note: `-D clippy::mem-unsafe-functions` implied by `-D warnings`

error: use of potentially dangerous memory manipulation function
  --> $DIR/mem_unsafe_functions.rs:30:9
   |
LL |         dont_use();
   |         ^^^^^^^^
   |
   = help: consider using its safe version

error: use of potentially dangerous memory manipulation function
  --> $DIR/mem_unsafe_functions.rs:31:9
   |
LL |         del_mem();
   |         ^^^^^^^
   |
   = help: consider using its safe version

error: use of potentially dangerous memory manipulation function
  --> $DIR/mem_unsafe_functions.rs:32:9
   |
LL |         memcpy();
   |         ^^^^^^
   |
   = help: consider using its safe version

error: aborting due to 4 previous errors

